CMakeLists.txt

```
@@ -233,12 +241,17 @@ endif()
 # project that we don't actually depend on. Further some of those parts
 # do not even compile on all platforms.
 if (TORCH_MLIR_ENABLE_STABLEHLO)
-  set(STABLEHLO_BUILD_EMBEDDED ON)
-  set(STABLEHLO_ENABLE_BINDINGS_PYTHON ON)
-  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo
-    ${CMAKE_CURRENT_BINARY_DIR}/stablehlo
-    EXCLUDE_FROM_ALL)
-  include_directories(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo)
+  if (NOT "${TORCH_MLIR_EXTERNAL_STABLEHLO_DIR}" STREQUAL "")
+    # Only include directories. It is assumed that stablehlo targets are made available by external project.
+    include_directories(${TORCH_MLIR_EXTERNAL_STABLEHLO_DIR})
+  else()
```

Actually
```suggestion
  # Only configure StableHLO if it isn't provided from a top-level project
  if ("${TORCH_MLIR_EXTERNAL_STABLEHLO_DIR}" STREQUAL "")
```
should be sufficient. In that case I would probably replace the string with a boolean as it seems that the string isn't used elsewhere.

Depending on how the super project is set up (and pulls in torch-mlir and stablehlo) we could consider to add a `TORCH_BUILD_EMBEDDED` option, as some parts in the current out-of tree build config might be in responsibility of the super project as well.

---

I have made suggested changes and added a boolean variable `TORCH_MLIR_USE_EXTERNAL_STABLEHLO`, set to OFF by default. As far as I can see, when `TORCH_MLIR_OUT_OF_TREE_BUILD` is enabled, MLIR package is searched by top level cmake.  Top level project can take care of this by passing `MLIR_DIR` and `LLVM_DIR` make options.